magicblock-committor-service/src/tasks/mod.rs

```diff
@@ -85,6 +85,7 @@ dyn_clone::clone_trait_object!(BaseTask);
 pub struct CommitTask {
     pub commit_id: u64,
     pub allow_undelegation: bool,
+    pub commit_diff: bool,
```

Comment:
The task of committing via diff should be introduced as a new variant of `ArgsTask` and `BufferTask`
```
#[derive(Clone)]
pub enum ArgsTaskType {
    Commit(CommitTask),
    CommitWithBuffer(CommitTask),
    Finalize(FinalizeTask),
    Undelegate(UndelegateTask), // Special action really
    BaseAction(BaseActionTask),
}
```

Now to avoid confusion we could also rename `ArgsTaskType::Commit` -> `ArgsTaskType::CommitWithState`